pmlc/runtime/runtime.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020, Intel Corporation
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <stdexcept>
+
+namespace pmlc::runtime {
+
+// Device represents a PlaidML device, capable of evaluating a PlaidML program.
+class Device {
+public:
+  // Returns the current Device, local to the current thread.
+  //
+  // If there is no current device, or the current device is of an unexpected
+  // type, an exception will be thrown.
+  //
+  // This is used to smuggle the current device through to runtimes that are
```

Comment:
I think we should pass the `runtime` instance to the LLVM code as the first void* pointer to the program. The `runtime` instance should be instantiated with the `device_id`.